File: kt/abc083_b.kt
fun main(args: Array<String>) {
  val (n, a, b) = readLine()!!.split(" ").map{ it.toInt() }
  // 1以上n以下の整数に各桁の和がa以上b以下になるか」でfilterをかけ、最後にそれらの和をsum()で返す
  var ans = (1..n).filter{ check(it.toString(), a, b) }.sum()
  println(ans)
}

fun check(s: String, a: Int, b: Int): Boolean {
  var sum = 0
  // まず対象の整数を文字列に変換し、1文字ずつ「'0'で引いたものを足し合わせていく」という操作をし、各文字はchar型として扱われているので、'0'で引くとASCIIコード的に該当する数字と同じものが得られる
  for (i in s.indices) sum += s[i] - '0'
  if (sum in a..b) return true else return false
}

File: cpp/abc083_b.cpp
#include<iostream>
using namespace std;

// 桁和を求める関数
int digit_sum(int n) {
  int sum = 0;
  while (n > 0)
  {  // nが0になるまで
    sum += n % 10;

    // C++ での割り算は小数点以下は切り捨てになる
    n /= 10;
  }
  return sum;
}

// 整数Nが与えられたら、桁和sumを返すプログラム
int main()
{
  int N, A, B; cin >> N >> A >> B;
  int ans = 0;
  for (int i = 1; i <= N; i++) {
    int sum = digit_sum(i);
    if (A <= sum && sum <= B)
    {
      ans += i;
    }
  }
  cout << ans << endl;
  return 0;
}
